src/overlays/actors/ovl_Bg_Crace_Movebg/z_bg_crace_movebg.c

```diff
@@ -44,136 +51,151 @@ const ActorInit Bg_Crace_Movebg_InitVars = {
     }
-    return returnVal;
+
+    return ret;
 }
 
-void func_80A70970(BgCraceMovebg* this, PlayState* play) {
-    this->unk164 = 0.0f;
-    this->unk160 = 0.0f;
-    this->actionFunc = func_80A7099C;
+void BgCraceMovebg_OpeningDoor_SetupIdle(BgCraceMovebg* this, PlayState* play) {
+    this->targetDoorHeight = 0.0f;
+    this->doorHeight = 0.0f;
     this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y;
+    this->actionFunc = BgCraceMovebg_OpeningDoor_Idle;
 }
 
-void func_80A7099C(BgCraceMovebg* this, PlayState* play) {
-    if (Flags_GetSwitch(play, BGCRACEMOVEBG_GET_7F0(&this->dyna.actor))) {
-        func_80A709E4(this, play);
+/**
+ * Wait for this door's switch flag to be set, then start opening.
+ */
+void BgCraceMovebg_OpeningDoor_Idle(BgCraceMovebg* this, PlayState* play) {
+    if (Flags_GetSwitch(play, BG_CRACE_MOVEBG_GET_SWITCH_FLAG(&this->dyna.actor))) {
+        BgCraceMovebg_OpeningDoor_SetupOpen(this, play);
     }
 }
 
-void func_80A709E4(BgCraceMovebg* this, PlayState* play) {
-    this->actionFunc = func_80A70A08;
-    this->unk164 = 180.0f;
+void BgCraceMovebg_OpeningDoor_SetupOpen(BgCraceMovebg* this, PlayState* play) {
+    this->targetDoorHeight = 180.0f;
+    this->actionFunc = BgCraceMovebg_OpeningDoor_Open;
 }
 
-void func_80A70A08(BgCraceMovebg* this, PlayState* play) {
+/**
+ * Silde open, then do nothing.
+ */
+void BgCraceMovebg_OpeningDoor_Open(BgCraceMovebg* this, PlayState* play) {
     func_800B9010(&this->dyna.actor, NA_SE_EV_STONEDOOR_OPEN_S - SFX_FLAG);
-    Math_SmoothStepToF(&this->unk160, this->unk164, 2.0f, this->unk16C, 0.01f);
-    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + this->unk160;
-    if (this->unk160 == this->unk164) {
-        func_80A70A84(this, play);
+    Math_SmoothStepToF(&this->doorHeight, this->targetDoorHeight, 2.0f, this->openSpeed, 0.01f);
+    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y + this->doorHeight;
+    if (this->doorHeight == this->targetDoorHeight) {
+        BgCraceMovebg_OpeningDoor_SetupDoNothing(this, play);
     }
 }
 
-void func_80A70A84(BgCraceMovebg* this, PlayState* play) {
-    this->actionFunc = func_80A70A9C;
+void BgCraceMovebg_OpeningDoor_SetupDoNothing(BgCraceMovebg* this, PlayState* play) {
+    this->actionFunc = BgCraceMovebg_OpeningDoor_DoNothing;
 }
 
-void func_80A70A9C(BgCraceMovebg* this, PlayState* play) {
+void BgCraceMovebg_OpeningDoor_DoNothing(BgCraceMovebg* this, PlayState* play) {
 }
 
 void BgCraceMovebg_Destroy(Actor* thisx, PlayState* play) {
     BgCraceMovebg* this = THIS;
 
     DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
-    if (!(this->unk170 & 0x8)) {
-        switch (BGCRACEMOVEBG_GET_F(&this->dyna.actor)) {
-            case 0:
-            case 2:
-                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx));
-                Flags_UnsetSwitch(play, BGCRACEMOVEBG_GET_7F0(thisx) + 1);
+    if (!(this->flags & BG_CRACE_MOVEBG_FLAG_ALREADY_LOADED)) {
```

Comment:
We usually prefer something like `stateFlags` to avoid confusion with the actual actor `flags`.